tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/Simple.fs

```diff
@@ -126,4 +126,109 @@ match CaseA with
         |> withDiagnostics [
                 Warning 104, Line 10, Col 7, Line 10, Col 12, "Enums may take values outside known cases. For example, the value 'CaseB (enum<E> (0))' may indicate a case not covered by the pattern(s)."
                 Warning 25, Line 10, Col 7, Line 10, Col 12, "Incomplete pattern matches on this expression. For example, the value 'CaseA' may indicate a case not covered by the pattern(s)."]
-   
\ No newline at end of file
+   
+    [<Fact>]
+    let ``Pattern matching on a union case, single error, uses only pattern range`` () = 
+        Fsx """
+type Stuff =
+    | A
+    | B
+    | C
+
+let x v = 
+    match v with
+    | A -> ""
+    | A -> ""
+    | C -> ""
+        """
+        |> withOptions ["--test:ErrorRanges"]     
+        |> typecheck
+        |> shouldFail
+        |> withDiagnostics [
+            (Warning 25, Line 8, Col 11, Line 8, Col 12, "Incomplete pattern matches on this expression. For example, the value 'B' may indicate a case not covered by the pattern(s).")
+            (Warning 26, Line 10, Col 7, Line 10, Col 8, "This rule will never be matched")
+        ]
+    
+    [<Fact>]
+    let ``Pattern matching on a union case with when clause, single error, uses only pattern range`` () = 
+        Fsx """
+type Stuff =
+    | A
+    | B
+    | C
+
+let x v z = 
+    match v with
+    | A -> ""
+    | A when z > 5 -> ""
+    | C -> ""
+        """
+        |> withOptions ["--test:ErrorRanges"]     
+        |> typecheck
+        |> shouldFail
+        |> withDiagnostics [
+            (Warning 25, Line 8, Col 11, Line 8, Col 12, "Incomplete pattern matches on this expression. For example, the value 'B' may indicate a case not covered by the pattern(s).")
+            (Warning 26, Line 10, Col 7, Line 10, Col 8, "This rule will never be matched")
+        ]
+        
+    [<Fact>]
+    let ``Pattern matching on a union case with binding, single error, uses only pattern range`` () = 
+        Fsx """
+type Stuff =
+    | A
+    | B
+    | C
+
+let x v = 
+    match v with
+    | A -> ""
+    | A as foo -> ""
+    | C -> ""
+        """
+        |> withOptions ["--test:ErrorRanges"]     
+        |> typecheck
+        |> shouldFail
+        |> withDiagnostics [
+            (Warning 25, Line 8, Col 11, Line 8, Col 12, "Incomplete pattern matches on this expression. For example, the value 'B' may indicate a case not covered by the pattern(s).")
+            (Warning 26, Line 10, Col 12, Line 10, Col 15, "This rule will never be matched")
```

Comment:
Shouldn't this actually start earlier? As of now, this would underline "foo" but for me it would make more sense to underline "A as foo" here.

Reply:
Yeah I mean I just wonder if the current change makes things clearer or (accidentally) more obscure.
Maybe you can quickly go through the random [patterns in this list](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/pattern-matching#remarks) and see if many of them would have similar problems.

But if you don't notice anything dramatic, feel free to resolve this and build on top of in a followup.

Reply:
My main goal was to ensure we had 2 different error ranges. As before it, 2 errors were overlapping, and depending on where you hover, it will show 1 or 2 errors. Happy to abandon this PR and look at this, trying to fix the issues mentioned. 

But overall, I think it is an improvement. But "Your house your rules" :)

Reply:
Disabled the auto-merge to tests few more cases